pkgs/by-name/sl/slimevr/default.nix

```diff
@@ -0,0 +1,217 @@
+    pname = "${pname}-server";
+    inherit version src;
+
+    mitmCache = gradle.fetchDeps {
+      inherit pname;
+      data = ./deps.json;
+    };
+
+    # Patch out use of grgit. Upstream code expects to be in a Git repo.
+    patches = [
+      ./no-grgit.patch
+    ];
+
+    postPatch = ''
+      # Manually apply program version, since Git can't be accessed.
+      substituteInPlace server/desktop/build.gradle.kts \
+        --replace-fail 'val gitVersionTag = ""' 'val gitVersionTag = "v${version}"'
+
+      # Disable Android, so its files don't have to be patched.
+      substituteInPlace settings.gradle.kts \
+        --replace-fail 'include(":server:android")' ""
+    '';
+
+    nativeBuildInputs = [
+      gradle
+      makeWrapper
+    ];
+
+    # this is required for using mitm-cache on Darwin
+    __darwinAllowLocalNetworking = true;
+
+    gradleFlags = [ "-Dorg.gradle.java.home=${jdk17_headless}" ];
+
+    gradleBuildTask = "shadowJar";
+
+    doCheck = true;
+
+    installPhase = ''
+      runHook preInstall
+
+      install -Dm644 server/desktop/build/libs/slimevr.jar $out/share/slimevr/slimevr.jar
+      makeWrapper ${jdk17_headless}/bin/java $out/bin/slimevr-server \
+          --add-flags "${serverJavaOptions}" \
+          --add-flags "-jar $out/share/slimevr/slimevr.jar"
+
+      runHook postInstall
+    '';
+
+    meta = sharedMeta // {
+      mainProgram = "slimevr-server";
+    };
+  });
+in
+{
+  slimevr-server = server;
+  slimevr = rustPlatform.buildRustPackage {
+    inherit version pname src;
+
+    buildAndTestSubdir = "gui/src-tauri";
+
+    # `hyper-tls` and `rustls-pemfile` aren't in the upstream lockfile, but are in
+    # Tauri's. This lock file is the upstream one with both of those added.
+    # `hper-tls` and `rustls-pemfile` are optional dependencies of `reqwest`.
+    cargoLock = {
+      lockFile = ./Cargo.lock;
+    };
+
+    pnpmDeps = pnpm_9.fetchDeps {
+      pname = "${pname}-pnpm-deps";
+      inherit version src;
+      hash = "sha256-5IqIUwVvufrws6/xpCAilmgRNG4mUGX8NXajZcVZypM=";
+    };
+
+    nativeBuildInputs = [
+      nodejs
+      pnpm_9.configHook
+      cargo-tauri.hook
+      pkg-config
+      wrapGAppsHook3
+      makeWrapper
+    ];
+
+    buildInputs =
+      [
+        openssl.dev
+        gst_all_1.gstreamer
+        gst_all_1.gst-plugins-base
+        gst_all_1.gst-plugins-good
+        gst_all_1.gst-plugins-bad
+      ]
+      ++ lib.optionals stdenv.isLinux [
+        glib-networking
+        libayatana-appindicator
+        webkitgtk_4_1
+      ];
+
+    # Upstream code expects to be in a Git repo.
+    patches = [ ./gui-no-git.patch ];
+
+    postPatch =
+      ''
+        rm Cargo.lock
+        cp --no-preserve=mode ${./Cargo.lock} Cargo.lock
+
+        # Manually apply program version in place of Git autodetection.
+        substituteInPlace gui/vite.config.ts \
+          --replace-fail 'const versionTag = "";' 'const versionTag = "v${version}";'
```

Comment:
I think you can just include this in the patch, but let the line be
```
const versionTag = "@version@";
```

Then wrap the patch file with `pkgs.replaceVars`:
```nix
  patches = [
    (replaceVars ./gui-no-git.patch {
      inherit version;
    })
  ]
```